src/redux/books/books.js

```diff
@@ -0,0 +1,29 @@
+const ADD_BOOK = 'bookStore/books/ADD_BOOK';
+const REMOVE_BOOK = 'bookStore/books/REMOVE_BOOK';
+
+const initialState = [];
+
+export const addBook = (playload) => ({
+  type: ADD_BOOK,
+  playload,
+});
+
+export const removeBook = (playload) => ({
+  type: REMOVE_BOOK,
+  playload,
+});
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      return [...state, action.playload];
+
+    case REMOVE_BOOK:
+      return state.filter((book) => book.id !== action.playload);
+
+    default:
+      return state;
+  }
+};
```

Comment:
Here I don't understand what it's wrong exactly because as I see all the requirements are implemented.

1. I write my book's actions and reducer (in /src/redux/books/books.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).
2. My action types for adding and removing a book are defined.
3. I Set the initial state to be an empty array of books.
4. I Exported action Creators for my actions.
5. And I wrote my reducer and exported it as default: - I define state changes for the actions that I created
                                                                                    - And in case of unknown action, it returns the current state.

Reply:
Maybe if you give me a clear explanation about it I can follow and do the change as needed.

Thanks :pray:

Reply:
Hello, fair enough I will explain.

The duck patterns is not just how to make the files in order is also the codes inside.
So instead of using const.... You need to use export default on every single one of them.

Reply:
Hello, so thanks for the explanation.
In duck patterns, they just use export default for reducer and I did this, and for other functions, they use just export and I did this, the difference between my code and their code is it's just a function, I prefer to use the arrow functions but they use the normal functions. So I'm supposed to use exactly the same code as them?? I can't even put a little change??